src/BitStringAddresses/fermifs.jl

```diff
@@ -0,0 +1,216 @@
+    S = typeof(BitString{M}(0))
+    return FermiFS{N,M,S}(SVector{M}(onr))
+end
+function FermiFS{N}(onr::Union{SVector{M},Tuple{M}}) where {N,M}
+    return FermiFS{N,M}(onr)
+end
+function FermiFS(onr::Union{AbstractVector,Tuple})
+    M = length(onr)
+    N = sum(onr)
+    return FermiFS{N,M}(onr)
+end
+
+function Base.show(io::IO, f::FermiFS{N,M}) where {N,M}
+    print(io, "FermiFS{$N,$M}(", tuple(onr(f)...), ")")
+end
+
+Base.bitstring(a::FermiFS) = bitstring(a.bs)
+Base.isless(a::FermiFS, b::FermiFS) = isless(a.bs, b.bs)
+Base.hash(a::FermiFS,  h::UInt) = hash(a.bs, h)
+Base.:(==)(a::FermiFS, b::FermiFS) = a.bs == b.bs
+num_occupied_modes(::FermiFS{N}) where {N} = N
+find_mode(::FermiFS, i) = i
+
+function near_uniform(::Type{FermiFS{N,M}}) where {N,M}
+    return FermiFS([fill(1, N); fill(0, M - N)])
+end
+
+onr(a::FermiFS) = SVector(m_onr(a))
+
+"""
+    FermiOccupiedModes
+
+Iterator over occupied modes in address. See [`occupied_modes`](@ref).
+"""
+struct FermiOccupiedModes{N,S}
+    bs::S
+end
+
+occupied_modes(a::FermiFS{N,<:Any,S}) where {N,S} = FermiOccupiedModes{N,S}(a.bs)
+
+Base.length(::FermiOccupiedModes{N}) where {N} = N
+Base.eltype(::FermiOccupiedModes) = Int
+
+function Base.iterate(o::FermiOccupiedModes)
+    c = 0
+    chunk = o.bs.chunks[end]
+    while iszero(chunk)
+        c += 1
+        chunk = o.bs.chunks[end - c]
+    end
+    zeros = trailing_zeros(chunk % Int)
+    return iterate(o, (chunk >> (zeros % UInt64), c * 64 + zeros, c))
+end
+function Base.iterate(o::FermiOccupiedModes, st)
+    chunk, index, c = st
+    while iszero(chunk)
+        c += 1
+        c == num_chunks(o.bs) && return nothing
+        chunk = o.bs.chunks[end - c]
+        index = c * 64
+    end
+    zeros = trailing_zeros(chunk % Int)
+    index += zeros
+    chunk >>= zeros
+    return index + 1, (chunk >> 1, index + 1, c)
+end
+
+function Base.iterate(o::FermiOccupiedModes{<:Any,<:BitString{<:Any,1,T}}) where {T}
+    chunk = o.bs.chunks[end]
+    zeros = trailing_zeros(chunk % Int)
+    return iterate(o, (chunk >> (zeros % T), zeros))
+end
+function Base.iterate(o::FermiOccupiedModes{<:Any,<:BitString{<:Any,1,T}}, st) where {T}
+    chunk, index = st
+    iszero(chunk) && return nothing
+    chunk >>= 0x1
+    index += 1
+    zeros = trailing_zeros(chunk % Int)
+    return index, (chunk >> (zeros % T), index + zeros)
+end
+
+function find_occupied_mode(a::FermiFS, i)
+    for k in occupied_modes(a)
+        i -= 1
+        i == 0 && return k
+    end
+    return 0
+end
+
+@inline function m_onr(a::FermiFS{<:Any,M}) where {M}
+    result = zero(MVector{M,Int32})
+    j = num_chunks(a.bs)
+    @inbounds for mode in occupied_modes(a)
+        result[mode] = 1
+    end
+    return result
+end
+
+function is_occupied(a::FermiFS{<:Any,M,S}, mode) where {M,T,S<:BitString{<:Any,1,T}}
+    @boundscheck 1 ≤ mode ≤ M || throw(BoundsError(a, mode))
+    return a.bs.chunks[1] & (T(1) << (mode - 1) % T) > 0
+end
```

Comment:
It might make sense to make a separate PR that makes interacting with addresses more convenient. We could try something like this, and also include implementations of multi-particle operators, etc.